README.md

```diff
@@ -14,35 +17,147 @@ checks syntax, variable references, and types, and warns about common JavaScript
+
+#### See the [Google Developers Site](https://developers.google.com/closure/compiler/docs/gettingstarted_app) for documentation including instructions for running the compiler from the command line.
+
+### NodeJS API
+You can access the compiler in a JS program by importing
+`google-closure-compiler`:
 
+```javascript
+import closureCompiler from 'google-closure-compiler';
+const { compiler } = closureCompiler;
+
+new compiler({
+  js: 'file-one.js',
+  compilation_level: 'ADVANCED'
+});
+```
+
+This package will provide programmatic access to the native Graal binary in most
+cases, and will fall back to the Java version otherwise.
+
+#### Please see the [closure-compiler-npm](https://github.com/google/closure-compiler-npm/tree/master/packages/google-closure-compiler) repository for documentation on accessing the compiler in JS.
+
+## Compiling Multiple Scripts
+
+If you have multiple scripts, you should compile them all together with one
+compile command.
+
+```bash
+google-closure-compiler in1.js in2.js in3.js --js_output_file out.js
+```
+
+You can also use minimatch-style globs.
+
+```bash
+# Recursively include all js files in subdirs
+google-closure-compiler 'src/**.js' --js_output_file out.js
+
+# Recursively include all js files in subdirs, excluding test files.
+# Use single-quotes, so that bash doesn't try to expand the '!'
+google-closure-compiler 'src/**.js' '!**_test.js' --js_output_file out.js
+```
+
+The Closure Compiler will concatenate the files in the order they're passed at
+the command line.
+
+If you're using globs or many files, you may start to run into problems with
+managing dependencies between scripts. In this case, you should use the [Closure
+Library](https://developers.google.com/closure/library/). It contains functions
+for enforcing dependencies between scripts, and Closure Compiler will re-order
+the inputs automatically.
+
+## Getting Help
+
+1.  Post in the [Closure Compiler Discuss
+    Group](https://groups.google.com/forum/#!forum/closure-compiler-discuss).
+2.  Ask a question on [Stack
+    Overflow](https://stackoverflow.com/questions/tagged/google-closure-compiler).
+3.  Consult the [FAQ](https://github.com/google/closure-compiler/wiki/FAQ).
+
+## Building the Compiler
+
+To build the compiler yourself, you will need the following:
+
+| Prerequisite | Description |
+| --- | --- |
+| [Java 8 or later](https://java.com) | Used to compile the compiler's source code. |
+| [Git](https://git-scm.com/) | Used by Bazel to download dependencies. |
+| [Bazel](https://docs.bazel.build/versions/master/install.html) | Used to build the various compiler targets. |
+
+### Installing Bazel
+
+#### Linux / MacOS
+All other required tools should be available in a [standard
+distribution](https://docs.bazel.build/versions/master/install.html).
+
+#### Windows
 On Windows, building closure-compiler additionally requires MSYS2, and a zip
 command-line executable (compatible with the one in linux).
```

Comment:
I think this is also stale https://docs.bazel.build/versions/master/windows.html#running-bazel-msys2-shell-vs-command-prompt-vs-powershell

I'm not even sure it's worthwhile to differentiate OSes here, but it's tough to say because I don't think anyone on the main team ever builds on Windows.